db/db_impl/db_impl_files.cc

```diff
@@ -220,10 +216,7 @@ void DBImpl::FindObsoleteFiles(JobContext* job_context, bool force,
         continue;
```

Comment:
the `log_sync_cv_` may wait here while holding the mutex

Reply:
I think `condvar::Wait` will implicitly release mutex (a mutex is paired with it when constructed).

Reply:
yes, it will release the pairing mutex `log_write_mutex_`, but not `mutex_`

Reply:
Good catch...  I'm thinking about how to solve this problem....

Reply:
I'm not sure whether it is still correctly to release `mutex_` here because if some other thread calling `FindObsoleteFiles` but this thread is only finish half of its task.

Reply:
I do not think it is a terrible case because this method will still block `mutex_` when other thread is synchronizing wal log file.

Reply:
`mutex_` is released before, I don't see a problem here. But you can't simply re-acquire the `mutex` after `Wait()`. That will break the lock order constraint.

Reply:
OK I will unlock `mutex_` before `FindObsoleteLogFiles`  method. PTAL